src/Controller/Component/DataTablesComponent.php

```diff
@@ -206,4 +251,38 @@ private function _addCondition($column, $value, $type = 'and')
             $this->config('matching', [$association => $condition]); // merges
         }
     }
+
+    /**
+     * Get comparison operator by entity and column name.
+     *
+     * @param  string $column Database column
+     * @return string         Database comparison operator
+     */
+    protected function _getComparison($column)
+    {
+        $config = new Collection($this->config('comparison'));
+        $tableName = $entity = $this->_tableName;
+        $columnName = $column;
+
+        // Attempt to find the table name
+        if (($pos = strpos($column, '.')) !== false) {
+            $entity = substr($column, 0, $pos);
+            $tableName = TableRegistry::get($entity)->table();
+            $columnName = substr($column, $pos + 1);
+        }
+
+        // Lookup the controller config for the comparison operator
+        $userConfig = $config->filter(function ($item, $key) use ($entity, $columnName) {
+            return strtolower($key) === strtolower(sprintf('%s.%s', $entity, $columnName));
+        });
+
+        if (!$userConfig->isEmpty()) {
+            return $userConfig->first();
+        }
+
+        // Lookup the application config for the comparison operator
+        $columnDesc = $this->_collection->describe($tableName)->column($columnName);
```

Comment:
I think it should be better here to use `TableRegistry::get($tableName)->schema()->column($columnName);`. So we don't need the 'comparison_datasource' parameter and the $_collection member variable. Agree?